projects/igniteui-angular/src/lib/grids/grid/grid-toolbar.spec.ts

```diff
@@ -227,105 +238,105 @@ describe('IgxGrid - Grid Toolbar #grid', () => {
         expect(getExportOptions(fixture).length).toBe(2);
 
         getExportButton(fixture).nativeElement.click();
-
-        // expect(getOverlay()).toBe(null);
     });
 
-    it('testing excel export starting event (cancel)', (done) => {
+    it('testing excel export starting event (cancel)', () => {
         grid.showToolbar = true;
         grid.exportExcel = true;
         fixture.detectChanges();
 
-        getExportButton(fixture).nativeElement.click();
-        const exportExcelButton = getOverlay(fixture).querySelector('li#btnExportExcel');
-
+        spyOn(grid.onToolbarExporting, 'emit').and.callThrough();
         grid.onToolbarExporting.pipe(first()).subscribe((args) => {
-            expect(args.grid).not.toBe(null);
-            expect(args.exporter).not.toBe(null);
-            expect(args.options).not.toBe(null);
-            expect(args.options instanceof IgxExcelExporterOptions).toBeTruthy();
-            expect(args.cancel).toBe(false);
             args.cancel = true;
-            done();
         });
+        spyOn(grid.toolbar.excelExporter, 'export');
 
+        getExportButton(fixture).nativeElement.click();
```

Comment:
You can change it to getExportButton(fixture).triggerEventHandler('click', new Event('click')